### PR TITLE
Feature/custom fields

### PR DESCRIPTION
**Pull request by piotr-galas**

### CODE REVIEW

**Comment by piotr-galas** [comment]
@himynameisjonas  I am not sure about it.  From the user perspective, it does not make sense, but API is used by developers. I think it is much easier for us to create API 1:1 as our database structure.

I do not know how to call it better. `CustomFieldValue` ?

**Comment by himynameisjonas** [comment]
> @himynameisjonas I am not sure about it. From the user perspective, it does not make sense, but API is used by developers. I think it is much easier for us to create API 1:1 as our database structure.
> 
> I do not know how to call it better. `CustomFieldValue` ?

I don't think it makes sense for developers either. It should have been named something else in our app/db. It's not a picked custom field but, as you said, the value of a custom field. Don't know if `CustomFieldValue` is the best, but it is a lot better than the current name. The current name sounds like you pick what customfields that should be available, not storing the candidate's value for a specific field.
Maybe ask in Slack for suggestions, I know i've talked about this with someone/someones (I don't remember who) before.

File: source/includes/custom-fields/_index.md.erb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+There are 5 types of custom field as specified by the <code>field-type</code> attribute.
+
+
+* __CustomField::Text__   any kind of text
+
+* __CustomField::Email__  valid email address e.g `example@example.com`
+
+* __CustomField::Url__  valid url e.g `https://google.com`
+
+* __CustomField::Phone__ It is an number, only ineger values are allowed
```
Comment (richardjohansson) [inline source/includes/custom-fields/_index.md.erb]:
```suggestion
* __CustomField::Phone__ A phone number, can include any values, eg `+46 723 3223 123`
```
Phone field can include strings, right?

Comment (piotr-galas) [reply source/includes/custom-fields/_index.md.erb]: Good cath 👍

File: source/includes/custom_field_values/_index.md.erb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+There are 5 types of custom field as specified by the <code>field-type</code> attribute.
+
+
+* __CustomField::Text__   any kind of text
+
+* __CustomField::Email__  valid email address e.g `example@example.com`
+
+* __CustomField::Url__  valid url e.g `https://google.com`
+
+* __CustomField::Phone__ It is an number, only ineger values are allowed
```
Comment (richardjohansson) [inline source/includes/custom_field_values/_index.md.erb]: Same here

Comment (piotr-galas) [reply source/includes/custom_field_values/_index.md.erb]: Oh no, shame on me. I will review by myself again. sorry

File: source/includes/custom_field_values/_index.md.erb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    }
+  }
+```
+
+### Attributes
+
+Attribute       | Type    | Description
+----------------|---------|------------
+value           | string  | Mandatory. Depending on the type, format may change. Check the available types above"
+field-type      | string  | Read only. Tells of which type the picked-custom-field is.
```
Comment (himynameisjonas) [inline source/includes/custom_field_values/_index.md.erb]: no `picked` :)

**Comment by piotr-galas** [comment]
I have fixed few more small issues. I hope everything is correct now. Sorry for mistakes.
@richardjohansson @himynameisjonas

File: source/includes/_changelog.md.erb [hunk 4]
```diff
@@ -11,8 +11,12 @@ X-Api-Version : 20161108
 curl -X "GET" "https://api.teamtailor.com/v1/jobs" \
      -H "X-Api-Version: 20161108"
 ```
+### 2019-09-10 *Non breaking update*
+ * Added the [custom-fields](#custom-fields) resource.
+ * Added the [custom-field-value](#custom-field-value) resource.
```
Comment (richardjohansson) [inline source/includes/_changelog.md.erb]: We normally name our resources in plural in the docs.
```suggestion
 * Added the [custom-field-values](#custom-field-values) resource.
```

**Comment by piotr-galas** [comment]
Regards filtering what about 
`https://api.teamtailor.localhost/v1/candidates/116/custom-field-values` endpoint  Will it be sufficient?
@himynameisjonas

**Comment by himynameisjonas** [comment]
> Regards filtering what about
> `https://api.teamtailor.localhost/v1/candidates/116/custom-field-values` endpoint Will it be sufficient?
> @himynameisjonas

I can think of some scenarios that i think we should support (maybe we already have support for some of these):
1. Fetch a bunch a candidates, including all their custom field values.
2. [Maybe this endpoint](https://docs.teamtailor.com/#list-latest-hired-candidates) should be able to include custom fields as well
3. I can imagine that companies would like to filter and fetch candidates based on their values. Like fetch all candidates that has `true` as value for a specific boolean custom field

**Comment by piotr-galas** [comment]
1. It is included now. See picture below

img

2. The endpoint you mention fetch the job applications. I do not think we can add it there
3. It is not possible now. Not for this iteration. We do not have such logic.
We are preparing to implement it in the future. But for doing it we need to add intexes or move data to the elasticsearch. I think it is out of this scope.

I can maybe add  3 point to the potential improvements?
@himynameisjonas

**Comment by davidwennergren** [comment]
1. [BUG] When clicking the link in the reminder I got the the job list instead of the actual job ad. Maybe it's some misunderstanding since the copy says "Go to job listing". I think we normally only refer to a job like "Go to job" or "Show job ad". 

2. [OPINION] The thank you page looks very empty. Maybe add som emojis or graphical element? 
Teamtailor - Welcome to our career site 2019-10-15 09-42-13

3. [OPINION] I wrote the pitch so I'm to blame, but when I tried it seems a bit arbitrary to wait for 2 hours to send the first email. I think it's probably better to send it immediately, and then the user has it in the inbox if they need the link. Imagine they are on the bus and want to do the application in 30 minutes when they get home.

4 [QUESTION] Also I have not been able to test this, but the second email reminder  should only be sent if they candidate has not applied already.

5 [LEGAL] Since we store and show the candidate as a lead we might to have the usual GDPR disclaimer on the reminder page. Not sure about this need to talk Erik.

**Pull request merged**